internal/controller/spireserver_controller_mock_test.go

```diff
@@ -107,6 +107,16 @@ func TestSpireserverController(t *testing.T) {
 	}
 }
 
+func TestCheckTrustDomain(t *testing.T) {
+	validTrustDomain := checkTrustDomain("trustdomain")
+	assert.Equal(t, validTrustDomain, nil, "There should be no error with \"trustdomain\"")
+}
+
+func TestInvalidCheckTrustDomain(t *testing.T) {
```

Comment:
Similarly, for invalid trust domains, I would recommend passing in invalid DNS names or malformed DNS names too to test whether the function returns an error on those, as expected.